### PR TITLE
Default a prop before coercion

### PR DESCRIPTION
**Pull request by simplesmiler**
Proposed fix for https://github.com/vuejs/vue/issues/2606

### CODE REVIEW

**Comment by simplesmiler** [comment]
This is what I get for making commits from github onsite editor.
Looking into why it fails.

File: src/compiler/compile-props.js [hunk 1]
```diff
@@ -229,10 +229,10 @@ function makePropsLinkFn (props) {
 
 function processPropValue (vm, prop, rawValue, fn) {
   const isSimple = prop.dynamic && isSimplePath(prop.parentPath)
-  let value = coerceProp(prop, rawValue)
-  if (value === undefined) {
-    value = getPropDefaultValue(vm, prop)
+  if (rawValue === undefined) {
+    rawValue = getPropDefaultValue(vm, prop)
   }
+  let value = coerceProp(prop, rawValue)
```
Comment (posva) [inline src/compiler/compile-props.js]: If you want to keep compatibility with the old version you'll need to check again if the value is undefined, althought I don't see why your coerce fn would return undefined...

Comment (posva) [inline src/compiler/compile-props.js]: I don't see a reason to support returning undefined in coerce fn to set the value to the default. Just wanted to raise the point, we never know :smiley:

**Comment by simplesmiler** [comment]
@yyx990803 what do you think?

File: src/compiler/compile-props.js (hunk 1 above)
Comment (yyx990803) [inline src/compiler/compile-props.js]: in most cases you wouldn't use the two together actually... coerce can already take care of default values.

**Comment by JosephusPaye** [comment]
Thanks for the fix. I will use the workaround @simplesmiler  provided in #2606 until a new version is released.

**Pull request merged**